<launch>
	<arg name="veh" default="$(env VEHICLE_NAME)"/>
    <arg name="local" default="false" doc="true for running everything you possibly can on laptop. false for everything running on vehicle."/>
    <arg name="config" default="baseline" doc="Specify a config."/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />



		<!-- run forward_kinematics_node -->
	    <remap from="forward_kinematics_node/wheels_cmd" to="wheels_driver_node/wheels_cmd_executed" />

	    <include file="$(find dagu_car)/launch/forward_kinematics_node.launch">
	        <arg name="veh" value="$(arg veh)"/>
	        <arg name="local" value="$(arg local)"/>
	        <arg name="config" value="$(arg config)"/>
	    </include>
	<!-- End forward_kinematic_node -->


    <!-- Intersection navigation fake nodes -->
    <!-- stop at red line node -->
	<node ns="$(arg veh)" name="stop_node" pkg="intersection_navigation" type="stop_node.py" output="screen">
		<!-- Setup parameters -->
		<!-- TODO check if useless -->
		<param name="~veh" value="$(arg veh)"/>
	</node>


	<!-- Intersection localizer -->
     <remap from="~pose_img_in" to="intersection_navigation_node/pose_img_out" />
     <remap from="~pose_out" to="intersection_navigation_node/pose_in" />

 	<node ns="$(arg veh)" name="intersection_localizer_node" pkg="intersection_navigation" type="intersection_localizer_node.py" output="screen">
 		<!-- Setup parameters -->
 		<param name="~veh" value="$(arg veh)"/>
 	</node>
 <!-- End Intersection localizer -->


    <!-- Intersection navigation -->
    <remap from="~fsm" to="fsm_node/mode" />
    <remap from="~intersection_navigation_pose" to="lane_controller_node/intersection_navigation_pose" />
    <remap from="~img" to="camera_node/image/compressed" />
    <remap from="~cmds" to="forward_kinematics_node/velocity" />
    <remap from="~apriltags_out" to="apriltags_postprocessing_node/apriltags_out" />
  <!--  <remap from="~turn_type" to="random_april_tag_turns_node/turn_type"/> !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! -->
    <remap from="~in_lane" to="lane_filter_node/in_lane" />

	<node ns="$(arg veh)" name="intersection_navigation_node" pkg="intersection_navigation" type="intersection_navigation_node.py" output="screen">
	 <!-- Setup parameters -->
		<param name="~veh" value="$(arg veh)"/>
	</node>
	 <!-- End Intersection navigation -->
</launch>
